File: UsersForums/UsersForumService/UsersForumService/Models/Post.cs
using System.Text.Json.Serialization;
using UsersForumService.DAL.Entities;

namespace UsersForumService.Models
{
    public class Post
    {
        public int Id { get; set; }
        public int UserId { get; set; }
        public string? Name { get; set; }
        public string? Title { get; set; }
        public string? Content { get; set; }
        
        [JsonIgnore]
        public ICollection<int>? ResponsersIDs { get; set; }
        // Navigation Properties
        
        [JsonIgnore]
        public User? User { get; set; }  // Post creator
        
        [JsonIgnore]
        public ICollection<Response>? Responses { get; set; }  // List of responses
      
    }

    public class PostDto//Data Object
    {
        public int Id { get; set; }
        public int UserId { get; set; }
        public string? Name { get; set; }
        public string? Title { get; set; }
        public string? Content { get; set; }
        // Add additional fields as needed, but exclude complex ones like User and Responses
    }
}

File: UsersForum/UsersForumClient/users-forum-app/src/app/models/post.ts
import { User } from './user';
import {Response} from './response';

export class Post {
  id: number;
  userId: number;
  name: string;
  title?: string;
  content?: string;
  responsersIDs?: number[];
  isEditing?: boolean = false;

  // Navigation Properties
  user?: User;  // Post creator
  responses?: Response[];  // List of responses

  constructor(
    id: number,
    userId: number,
    name: string,
    user: User,
    title?: string,
    content?: string,
    responsersIDs?: number[],
    responses?: Response[]
  ) {
    this.id = id;
    this.userId = userId;
    this.name = name;
    this.title = title;
    this.content = content;
    this.responsersIDs = responsersIDs;
    this.user = user;
    this.responses = responses;
  }
}
